### ISSUE TITLE
Keep $ref

### ISSUE THREAD

**Issue by thim81** [thim81/openapi-format#71]
Investigate the option to leverage: https://www.npmjs.com/package/json-refs to format the OpenAPI documents and update the $ref values

**Comment by rngtng** [comment]
yes this would be great. in addition we make heavily use of `$ref` to other files what the best solution to tackle those? format one file after the other?

**Comment by rngtng** [comment]
Cool thanks a lot. I‘ll explore myself options too..

**Comment by thim81** [comment]
hi @rngtng 

With the release 1.24.0 of OpenAPI, we introduced the option to [split OpenAPI files in a file/folder structure](https://github.com/thim81/openapi-format?tab=readme-ov-file#splitting-the-openapi-document).

The `split` is a programmatically break-up of the OpenAPI in 1 main OpenAPI and per path and per component a separate file, with this kind of process flow.

```mermaid
graph TD
    A[Start] --> B[Load OpenAPI Split File]
    
    B --> C[Apply Bundling]
    C --> D[Apply Filtering]
    D --> E[Apply Sorting]
    
    E --> F{Split OpenAPI?}
    
    F -- Yes --> G[Split into Multiple Files]
    F -- No --> H[Save as Single File]
    
    G --> I[Save Modular Structure]
    H --> I[Save Final Output]
    
    I --> J[End]
```

You can give it a try 

```
npx openapi-format https://raw.githubusercontent.com/thim81/openapi-format/refs/heads/main/test/__utils__/train.yaml -o split/train.yaml --split
```

Let me know what you think of it.
